src/Tasks/PopulateSupportedArchitectures.cs

```diff
@@ -0,0 +1,200 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Xml;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Tasks.Deployment.ManifestUtilities;
+using Microsoft.Build.Utilities;
+
+#nullable enable
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Generates an application manifest or adds an entry to the existing one when PreferNativeArm64 property is true.
+    /// </summary>
+    [SupportedOSPlatform("windows")]
+    public sealed class PopulateSupportedArchitectures : TaskExtension
+    {
+        private const string supportedArchitectures = "supportedArchitectures";
+        private const string windowsSettings = "windowsSettings";
+        private const string SupportedArchitecturesValue = "amd64 arm64";
+        private const string asmv3Prefix = "asmv3";
+        private const string DefaultManifestName = "default.win32manifest";
+        private const string WindowsSettingsNamespace = "http://schemas.microsoft.com/SMI/2024/WindowsSettings";
+
+        private string _outputPath = string.Empty;
+        private string _generatedManifestFullPath = string.Empty;
+
+        /// <summary>
+        /// Path to the existing application manifest.
+        /// </summary>
+        public string? ApplicationManifestPath { get; set; }
+
+        /// <summary>
+        /// Intermediate output path.
+        /// </summary>
+        [Required]
+        public string OutputPath
+        {
+            get => _outputPath;
+            set => _outputPath = value ?? throw new ArgumentNullException(nameof(OutputPath));
+        }
+
+        /// <summary>
+        /// Returns path to the generated manifest.
+        /// </summary>
+        [Output]
+        public string ManifestPath
+        {
+            get => _generatedManifestFullPath;
+            private set => _generatedManifestFullPath = value;
+        }
+
+        public override bool Execute()
+        {
+            Debugger.Launch();
+            if (!string.IsNullOrEmpty(PathToManifest))
+            {
+                XmlDocument document = LoadManifest(PathToManifest);
+                XmlNamespaceManager xmlNamespaceManager = XmlNamespaces.GetNamespaceManager(document.NameTable);
+
+                ManifestValidationResult validationResult = ValidateManifest(document, xmlNamespaceManager);
+
+                switch (validationResult)
+                {
+                    case ManifestValidationResult.Success:
+                        PopulateSupportedArchitecturesElement(document, xmlNamespaceManager);
+                        SaveManifest(document);
+                        return true;
+                    case ManifestValidationResult.SupportedArchitecturesExists:
+                        return true;
```

Comment:
`ManifestPath` is not set in this case. Isn't it a problem since it's unconditionally assigned to a build property in `Microsoft.Common.CurrentVersion.targets`?

Reply:
it's a good catch, thank you :)